internal/service/alarms/internal/serviceconfig/serviceconfig.go

```diff
@@ -0,0 +1,300 @@
+									Command: []string{"psql"},
+									Args:    []string{"-f", fmt.Sprintf("%s/%s", cleanScriptDir, cleanScriptName)},
+									Env:     pgEnv,
+									Resources: corev1.ResourceRequirements{
+										Limits: corev1.ResourceList{
+											corev1.ResourceCPU:    resource.MustParse("100m"),
+											corev1.ResourceMemory: resource.MustParse("128Mi"),
+										},
+									},
+									VolumeMounts: []corev1.VolumeMount{
+										{
+											Name:      configMapName,
+											MountPath: cleanScriptDir,
+										},
+									},
+								},
+							},
+							Volumes: []corev1.Volume{{
+								Name: configMapName,
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: configMapName,
+										},
+									},
+								},
+							}},
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+// getPostgresImage dynamically collect the existing PG image to run psql from
+func getPostgresImage(ctx context.Context, c client.Client) (string, error) {
+	var deploymentList appsv1.DeploymentList
+	if err := c.List(ctx, &deploymentList, client.InNamespace(utils.DefaultNamespace),
+		client.MatchingLabels{"app": "postgres-server"}); err != nil {
+		return "", fmt.Errorf("list deployments: %w", err)
+	}
+
+	if len(deploymentList.Items) != 1 {
+		return "", fmt.Errorf("expected one postgres deployment, found %d", len(deploymentList.Items))
+	}
+
+	deployment := deploymentList.Items[0]
+	for _, container := range deployment.Spec.Template.Spec.Containers {
+		if container.Name == "server" {
+			if container.Image == "" {
+				return "", fmt.Errorf("postgres server container has empty image")
+			}
+			return container.Image, nil
+		}
+	}
+
+	return "", fmt.Errorf("server container not found in postgres deployment %s", deployment.Name)
+}
+
+// CleanupResources call this for cleanup of k8s resources
+func CleanupResources(ctx context.Context, hc client.Client) error {
+	// Delete the cronjob and wait for deletion
+	cronJob := &batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cleanupCronJobName,
+			Namespace: utils.DefaultNamespace,
+		},
+	}
+	if err := hc.Delete(ctx, cronJob); err != nil {
+		if !errors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete cleanup cronjob: %w", err)
+		}
+	}
+	// Wait for cronjob deletion
+	if err := wait.PollUntilContextTimeout(ctx, 2*time.Second, 2*time.Minute, true, func(ctx context.Context) (bool, error) {
+		err := hc.Get(ctx, types.NamespacedName{
+			Name:      cleanupCronJobName,
+			Namespace: utils.DefaultNamespace,
+		}, &batchv1.CronJob{})
+		if errors.IsNotFound(err) {
+			return true, nil
+		}
+		if err != nil {
+			return false, err //nolint:wrapcheck
+		}
+		return false, nil
+	}); err != nil {
+		return fmt.Errorf("failed waiting for cronjob deletion: %w", err)
+	}
+
+	// Delete the configmap and wait for deletion
+	configMap := &corev1.ConfigMap{
```

Comment:
Should we set the ownership of these new objects so that they can auto-deleted when the alarm-server is deleted or when the operator is deleted?

Reply:
Ah yeah good point this can be improved

Reply:
took ownership